src/box/lua/net_box.c

```diff
@@ -1162,7 +1162,13 @@ netbox_transport_send_and_recv(struct netbox_transport *transport,
 			const char *bufpos = transport->recv_buf.rpos;
 			const char *rpos = bufpos;
 			size_t len = mp_decode_uint(&rpos);
-			required = (rpos - bufpos) + len;
+			size_t size = rpos - bufpos;
+			if (size > SIZE_MAX - len) {
+				box_error_raise(ER_NO_CONNECTION,
+						"Data buffer size too large");
```

Comment:
Nit:
 - It's not "data buffer size" that is too large. It's "response size".
 - Let's decode the length into `uint64_t` because `mp_decode_uint` returns `uint64_t`.
 - Let's put `len` to the left to emphasize that we check `len`, not `size`.

```C
uint64_t len = mp_decode_uint(&rpos);
size_t size = rpos - bufpos;
if (len > SIZE_MAX - size) {
        box_error_raise(ER_NO_CONNECTION, "Response size is too large");
        return -1;
}
```

Reply:
Fixed.